sql/src/test/java/io/crate/analyze/AlterTableAddColumnAnalyzerTest.java

```diff
@@ -199,6 +199,26 @@ public void testAddNewNestedColumnToObjectArray() throws Exception {
     }
 
     @Test
+    public void testAddColumnToObjectTypeMaintainsObjectPolicy() throws Exception {
+        AddColumnAnalyzedStatement analysis = e.analyze(
+            "alter table users add column address['street'] string");
+        List<AnalyzedColumnDefinition> columns = analysis.analyzedTableElements().columns();
+        assertThat(columns.size(), is(2));
+
+        AnalyzedColumnDefinition address = columns.get(0);
+        assertThat(address.ident(), is(ColumnIdent.fromPath("address")));
+        assertThat(address.dataType(), is("object"));
+        assertThat(address.isParentColumn(), is(true));
+        assertThat(address.children().size(), is(1));
+        assertThat(address.getObjectType(), is ("strict"));
+
+        AnalyzedColumnDefinition street = address.children().get(0);
+        assertThat(street.ident(), is(ColumnIdent.fromPath("address.street")));
```

Comment:
if so I'd appreciate moving this in a separate test, otherwise I'd ask you in a few month again why this is relevant to this test ;)